### ISSUE TITLE
Open Source License scanning for Binary files

### ISSUE THREAD

**Issue by kiranravindran90** [armijnhemel/binaryanalysis-ng#207]
We installed BANG successfully and able to unpack binary files (.bin-flashwares).

But we are not able to locate result files for Open Source License scanning.
We see "LicenseIdentifierScanner.py" file in \src but not sure if it is called during unpacking or has to be called separately?

**Comment by armijnhemel** [comment]
BANG is split into two parts: unpacking and analysis. These two are separate. Currently the analysis part is under development and you should consider BANG as an unpacker for now. I will be merging the first analysis tools in  about a month.

**Comment by kiranravindran90** [comment]
Thanks for this info. Any update on the analysis part?

**Comment by armijnhemel** [comment]
> Thanks for this info. Any update on the analysis part?

There are a few analysis tools, but they are far from complete yet. See the file `README-analysis.md` in the top level directory to see what is there right now. More will be added soon: an identification mechanism based on Yara is currently in the works.

**Comment by kiranravindran90** [comment]
Oh ok. Thank you. I am specifically looking for Open source License scan. Any idea when that part would be available? A rough estimate should be more than enough, just to know how long to wait :)

**Comment by armijnhemel** [comment]
> Oh ok. Thank you. I am specifically looking for Open source License scan. Any idea when that part would be available? A rough estimate should be more than enough, just to know how long to wait :)

It depends on what you think by "open source license scan". Could you clarify your use case?

**Comment by kiranravindran90** [comment]
Well what I meant by that is - able to scan the Binary files for presence of any Open Source licenses like GPL, Apache etc.
Could be based on checking for License text in the Binary or so.

**Comment by armijnhemel** [comment]
> Well what I meant by that is - able to scan the Binary files for presence of any Open Source licenses like GPL, Apache etc. Could be based on checking for License text in the Binary or so.

This is actually not that difficult (quite trivial actually using a few YARA rules), but you should be aware that very few binaries actually incorporate license information.

**Comment by kiranravindran90** [comment]
Thank you for the suggestion & quick response. I shall check on that. Is there a plan to implement such a feature in BANG?

**Comment by armijnhemel** [comment]
> Thank you for the suggestion & quick response. I shall check on that. Is there a plan to implement such a feature in BANG?

Yes. As a matter of fact, I am working on a first version of YARA integration as we speak and that should be in BANG this week. Let me see what I can do :-)

**Comment by kiranravindran90** [comment]
Oh that's good. Thanks a lot  :)

**Comment by armijnhemel** [comment]
I have added some rules in the directory `maintenance/yara_rules`. You can compile these to YARA rules and then use YARA. I will soon write scripts that will be able to process output from BANG and use these rules as well.

**Comment by kiranravindran90** [comment]
Hello :) Are these scripts ready & available now? i.e. We can unpack Binaries & scan for Open source license information?
